src/components/graphs/ScatterGraph.vue

```diff
@@ -0,0 +1,499 @@
+    .style('background', 'linear-gradient(135deg, #f8fafc 0%, #f1f5f9 100%)')
+    .style('box-shadow', '0 4px 20px rgba(0, 0, 0, 0.1)');
+
+  const chartGroup = svg
+    .append('g')
+    .attr('transform', `translate(${margin.left},${margin.top})`);
+
+  // Title
+  svg
+    .append('text')
+    .attr('x', containerWidth / 2)
+    .attr('y', 30)
+    .attr('text-anchor', 'middle')
+    .style('font-size', '18px')
+    .style('font-weight', '600')
+    .style('fill', '#1e293b')
+    .text(props.title);
+
+  // Scales
+  xScale = d3
+    .scaleLinear()
+    .domain(
+      d3.extent(sortedData.value, (d: DataPoint) => d.year) as [number, number],
+    )
+    .range([0, width]);
+
+  const yExtent = d3.extent(sortedData.value, (d: DataPoint) => d.value) as [
+    number,
+    number,
+  ];
+  const yPadding = (yExtent[1] - yExtent[0]) * 0.1;
+  yScale = d3
+    .scaleLinear()
+    .domain([Math.max(0, yExtent[0] - yPadding), yExtent[1] + yPadding])
+    .range([height, 0]);
+
+  // Grid
+  const gridGroup = chartGroup.append('g').attr('class', 'grid');
+
+  gridGroup
+    .selectAll('.grid-line-horizontal')
+    .data(yScale.ticks(6))
+    .enter()
+    .append('line')
+    .attr('class', 'grid-line-horizontal')
+    .attr('x1', 0)
+    .attr('x2', width)
+    .attr('y1', (d) => yScale(d))
+    .attr('y2', (d) => yScale(d))
+    .style('stroke', '#e2e8f0')
+    .style('stroke-width', 1)
+    .style('opacity', 0.5);
+
+  gridGroup
+    .selectAll('.grid-line-vertical')
+    .data(xScale.ticks(Math.min(8, sortedData.value.length)))
+    .enter()
+    .append('line')
+    .attr('class', 'grid-line-vertical')
+    .attr('x1', (d) => xScale(d))
+    .attr('x2', (d) => xScale(d))
+    .attr('y1', 0)
+    .attr('y2', height)
+    .style('stroke', '#e2e8f0')
+    .style('stroke-width', 1)
+    .style('opacity', 0.5);
+
+  // Axes
+  chartGroup
+    .append('g')
+    .attr('class', 'x-axis')
+    .attr('transform', `translate(0,${height})`)
+    .call(
+      d3
+        .axisBottom(xScale)
+        .tickValues(sortedData.value.map((d) => d.year))
+        .tickFormat((d) => {
+          return containerWidth < 400 ? `'${String(d).slice(2)}` : String(d);
+        })
+        .tickSize(-10),
+    );
+
+  chartGroup
+    .append('g')
+    .attr('class', 'y-axis')
+    .call(d3.axisLeft(yScale).tickFormat(d3.format('.2s')).tickSize(-10));
+
+  // Style axes
+  svg
+    .selectAll('.x-axis, .y-axis')
+    .selectAll('path, line')
+    .style('stroke', '#64748b')
+    .style('stroke-width', 2);
+
+  svg
+    .selectAll('.x-axis, .y-axis')
+    .selectAll('text')
+    .style('fill', '#475569')
+    .style('font-size', '12px');
+
+  // Axis labels
+  chartGroup
+    .append('text')
+    .attr('transform', 'rotate(-90)')
+    .attr('y', 0 - margin.left)
+    .attr('x', 0 - height / 2)
+    .attr('dy', '2.5em')
+    .style('text-anchor', 'middle')
+    .style('font-size', '14px')
+    .style('font-weight', '600')
+    .style('fill', '#374151')
```

Comment:
See other comment - colors should always be in a constant to avoid magic strings and one off colors (I have no idea reading this hex code what color this is)